third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -214,14 +209,15 @@ static void moveUpTranspose(triton::FuncOp funcOp) {
 }
 
 // Schedule global load and local store ops for better GEMM performance.
-static void scheduleGlobalLoadLocalStore(triton::FuncOp funcOp) {
+static void scheduleGlobalLoadLocalStore(scf::ForOp forOp) {
   SmallVector<Operation *> moveOps;
-  // Move global loads early to prefetch. This may increase register pressure
-  // but it enables issuing global loads early.
-  funcOp.walk([&](triton::LoadOp op) { moveOps.push_back(op); });
   // Move local_stores early if dependence distance greater than one iteration.
   // Best perf on GEMM when these precede global loads.
-  funcOp.walk([&](ttg::LocalStoreOp op) { moveOps.push_back(op); });
+  forOp.walk([&](ttg::LocalStoreOp op) { moveOps.push_back(op); });
+
+  // Move global loads early to prefetch. This may increase register pressure
+  // but it enables issuing global loads early.
+  forOp.walk([&](triton::LoadOp op) { moveOps.push_back(op); });
```

Comment:
@sjw36, If I understood you correctly. We only need this change, right?

Reply:
This is really the only source change needed.